KmsClient.js

```diff
@@ -76,26 +76,29 @@ export class KmsClient {
       capability = _getRootZcapId({keystoreId});
     }
 
+    const headers = await _signCapabilityInvocation({
+      url, method: 'post', operation, capability, invocationSigner,
+      capabilityAction: 'generateKey'
+    });
+
     try {
-      // sign HTTP header
-      const headers = await signCapabilityInvocation({
-        url, method: 'post', headers: this.defaultHeaders,
-        json: operation, capability, invocationSigner,
-        capabilityAction: 'generateKey'
-      });
       // send request
       const {agent} = this;
       const result = await httpClient.post(url, {
         agent, headers, json: operation
       });
       return result.data;
     } catch(e) {
+      let errorMessage = 'Error generating key.';
       if(e.status === 409) {
-        const err = new Error('Duplicate error.');
-        err.name = 'DuplicateError';
-        throw err;
+        e.name = 'DuplicateError';
```

Comment:
Modifying the error doesn't seem right. Why not create a `DuplicateError` and set its `.cause = e`?

Reply:
@dlongley - `DuplicateError` is not something defined by Node. Are you suggesting creating a custom Error subclass, for use here?

Reply:
@dmitrizagidulin, no, just that the previous code was creating a new Error -- why did we stop doing that? I'm suggesting this:

```js
const err = new Error('Duplicate error.');
err.name = 'DuplicateError';
err.cause = e;
```

Instead of mutating `e` itself.

Reply:
Ah, sure thing.

Reply:
Refactored.